test/powershell/Modules/Microsoft.PowerShell.Utility/ConvertTo-Json.Tests.ps1

```diff
@@ -62,4 +62,38 @@ Describe 'ConvertTo-Json' -tags "CI" {
 
         @{ 'abc' = "'def'" } | ConvertTo-Json @params | Should -BeExactly $expected
     }
+
+    It "Should handle 'AutomationNull.Value' and 'NullString.Value' correctly" {
+        [ordered]@{
+            a = $null;
+            b = [System.Management.Automation.Internal.AutomationNull]::Value;
+            c = [System.DBNull]::Value;
+            d = [NullString]::Value
+        } | ConvertTo-Json -Compress | Should -BeExactly '{"a":null,"b":null,"c":null,"d":null}'
+
+        ConvertTo-Json -Compress ([System.Management.Automation.Internal.AutomationNull]::Value) | Should -BeExactly 'null'
+        ConvertTo-Json -Compress ([NullString]::Value) | Should -BeExactly 'null'
+
+        ConvertTo-Json -Compress @(
```

Comment:
just for testing slightly different code path, it might be good to make some of these 4 ` ConvertTo-Json -Compress` checks without the `-Compress` parameter.

Reply:
Updated to not use `-Compress` for 2 of them.